fhir-server/src/main/java/com/ibm/fhir/server/util/FHIRRestHelper.java

```diff
@@ -754,12 +754,13 @@ private Resource doRead(String type, String id, boolean throwExcOnNull, boolean
                 boolean lenient = HTTPHandlingPreference.LENIENT.equals(requestContext.getHandlingPreference());
 
                 // Determine if any non-general search parameters are specified
-                if (!queryParameters.keySet().stream().allMatch(k -> FHIRConstants.GENERAL_PARAMETER_NAMES.contains(k))) {
-                    FHIRSearchException se = SearchExceptionUtil.buildNewInvalidSearchException("Read only supports general search parameters.");
+                List<String> nonGeneralParams = queryParameters.keySet().stream().filter(k -> !FHIRConstants.GENERAL_PARAMETER_NAMES.contains(k)).collect(Collectors.toList());
+                for (String nonGeneralParam : nonGeneralParams) {
```

Comment:
Might be nice to show all the failing parms in the exception msg, since the way it's written you'll only get the first one before you throw the exception. Rather than a for loop, maybe a check if list is not empty, then update the exception msg to be "Search parameters " + nonGeneralParams + " not supported by read". It'll format the parm strings inside brackets. Then for the log msg, can use your prev msg since the exception will contain the failing parms.

Reply:
Yeah, I went this route since this is how all the current invalid search parameter checking works.  Which is, as soon as an invalid search parameter (or value) is found, it throws an exception for the single parameter and parsing ends.  I can have just this error work differently, but thought consistency would be better.

Reply:
I'm fine with leaving it the way you have it.